book.md

```diff
@@ -495,17 +492,160 @@ probably include, or be followed by, a section on the various sanitizers
 available (ASan, UBSan, etc).}
 
 ## Code reuse attacks
-\missingcontent{Discuss ROP, JOP, COOP and mitigations (ASLR, CFI etc)}
+
+In the early days of memory vulnerability exploitation, attackers could simply
+place shellcode\index{shellcode} of their choice in executable memory and jump
+to it. As executable-space protection became mainstream, attackers started to
```

Comment:
"executable-space protection". It seems that earlier in the book the term "non-executable stack" and similar are used, but not "executable-space protection". The term "executable-space protection" made me wonder what the exact definition of that would be. Would using "non-executable stack" here be less confusing for most readers?

Reply:
I was trying to avoid specifically mentioning the stack as the same can be said of the heap. I'll say "non-executable stack and heap" instead.